src/algo/depth-to-rgb-calibration/calibration-types.h

```diff
@@ -0,0 +1,202 @@
+    };
+
+        struct double3x3
+    {
+        /* double3x3(double arr[9])
+         {
+             for (auto i = 0; i < 3; i++)
+             {
+                 for (auto j = 0; j < 3; j++)
+                 {
+                     mat[i][j] = arr[i * 3 + j];
+                 }
+             }
+         }*/
+        double mat[3][3];
+        double3x3 transpose()
+        {
+            double3x3 res = { 0 };
+
+            for( auto i = 0; i < 3; i++ )
+            {
+                for( auto j = 0; j < 3; j++ )
+                {
+                    res.mat[i][j] = mat[j][i];
+                }
+            }
+            return res;
+        }
+
+        double3x3 operator*( const double3x3 & other )
+        {
+            double3x3 res = { 0 };
+
+            for( auto i = 0; i < 3; i++ )
+            {
+                for( auto j = 0; j < 3; j++ )
+                {
+                    double sum = 0;
+                    for( auto l = 0; l < 3; l++ )
+                    {
+                        sum += mat[i][l] * other.mat[l][j];
+                    }
+                    res.mat[i][j] = sum;
+                }
+            }
+            return res;
+        }
+
+        double3 operator*( const double3 & other )
+        {
+            double3 res = { 0 };
+
+            res.x = mat[0][0] * other.x + mat[0][1] * other.y + mat[0][2] * other.z;
+            res.y = mat[1][0] * other.x + mat[1][1] * other.y + mat[1][2] * other.z;
+            res.z = mat[2][0] * other.x + mat[2][1] * other.y + mat[2][2] * other.z;
+            return res;
+        }
+
+        std::vector< double > to_vector()
+        {
+            std::vector< double > res;
+            for( auto i = 0; i < 3; i++ )
+            {
+                for( auto j = 0; j < 3; j++ )
+                {
+                    res.push_back( mat[i][j] );
+                }
+            }
+            return res;
+        }
+    };
+
+    enum direction :uint8_t
+    {
+        //deg_111, //to be aligned with matlab (maybe should be removed later) 
+        deg_0, // 0, 1
+        deg_45, //1, 1
+        deg_90, //1, 0
+        deg_135, //1, -1
+        deg_180, //
+        deg_225, //
+        deg_270, //
+        deg_315, //
+        deg_none
+    };
+    enum svm_model :uint8_t
+    {
+        linear,
+        gaussian
+    };
+    struct translation
+    {
+        double t1;
+        double t2;
+        double t3;
+    };
+
+    struct matrix_3x3
+    {
+        double rot[9];
+
+        matrix_3x3 transposed() const
+        {
+            return { rot[0], rot[3], rot[6], 
+                rot[1], rot[4], rot[7],
+                rot[2], rot[5], rot[8] };
+        }
+    };
+
+    struct rotation_in_angles
+    {
+        double alpha;
+        double beta;
+        double gamma;
+
+        bool operator==(const rotation_in_angles& other)
+        {
+            return alpha == other.alpha && beta == other.beta && gamma == other.gamma;
```

Comment:
Using == for float types comparison is risky, better
```fabs(a-b) < std::numeric_limits<double>::epsilon()```

Reply:
I don't know where it's even used -- it's a valid concern though, just don't think now's the time.